src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.cs

```diff
@@ -186,11 +186,11 @@ public DependencyPropertyDetails GetPropertyDetails(DependencyProperty property)
 					_entries = entries = newEntries;
 				}
 
-				ref var propertyEntry = ref entries[offset + bucketRemainder];
-
-				if (propertyEntry == null)
+				var propertyEntry = entries[offset + bucketRemainder];
+				if (propertyEntry is null)
 				{
 					propertyEntry = new DependencyPropertyDetails(property, _ownerType, property == _dataContextProperty || property == _templatedParentProperty);
+					_entries[offset + bucketRemainder] = propertyEntry;
 
 					if (TryResolveDefaultValueFromProviders(property, out var value))
```

Comment:
Had to make this change because in `TryResolveDefaultValueFromProviders` the `KeyboardAcceleratorCollection` instance got created, which then leads to it setting its `Parent` to the `UIElement` instance which owns it - this then can cause `entries` to get resized, which means the original `ref` is no longer valid and all kinds of hell breaks loose. Because the `entries` collection size only increases here, it seemed safe to do it this way, but please let me know if you see some red herrings @jeromelaban @ebariche

Reply:
We'll likely need to make this change in a separate PR with specialized tests.